### ISSUE TITLE
PointNet: Deep Learning on Point Sets for 3D Classification and Segmentation

### ISSUE THREAD

**Issue by wrryu09** [GDSC-Ewha-4th/Study-paperReview#65]
## PointNet: Deep Learning on Point Sets for 3D Classification and Segmentation

Charles R. Qi, Hao Su, Kaichun Mo, Leonidas J. Guibas
; Proceedings of the IEEE Conference on Computer Vision and Pattern Recognition (CVPR), 2017, pp. 652-660

#
#
 we design a novel type of **neural network that directly consumes point clouds**, which well respects the permutation invariance of points in the input.

#
### introduction

point cloud는 불규칙한 포맷을 가져서 기존 연구들은 대부분 regular 3D voxel grids, collections of images로 변환해서 deep net architecture에 사용함 → renders data unnecessarily voluminous and causes issues

point clouds는 조합에서 오는 불규칙함, 메쉬의 복잡성을 피할 수 있는 simple, unified structure임

그러나 certain symmetrizations in the net computation 필요
#

PointNet is a unified architecture that directly takes point clouds as input and outputs either class labels for the entire input or per point segment/part labels for each point of the input.

Key : use of a single symmetric function, max pooling

네트워크는 point cloud 에서 중요한 점들을 고르는 기준을 학습하고 선택의 이유 encode. 마지막 단계에서 완전히 연결된 레이어는 이 학습된 최적의 값들을 global descriptor에 통합하고 이걸로 shape classification, shape segmentation 할 수 있음
#
key contributes

- We design a novel deep net architecture suitable for consuming unordered point sets in 3D;
- We show how such a net can be trained to perform 3D shape classification, shape part segmentation and scene semantic parsing tasks;
- We provide thorough empirical and theoretical analy- sis on the stability and efficiency of our method;
- We illustrate the 3D features computed by the selected neurons in the net and develop intuitive explanations for its performance.

point cloud is represented as a set of 3D points {$Pi|i=1,…,n$}, where each point $Pi$ is a vector of its (x,y,z)
#
![image](https://user-images.githubusercontent.com/98469609/204133394-fa896ebd-2161-49f7-9e58-4c93f0ac9c8d.png)

classification network과 segmentation network는 구조 많이 겹침
#
### Deep Learning on Point Sets
#
properties of Point Sets in $R^n$

- Unordered.
- Interaction among points.
- Invariance under transformations.
#
### PointNet Architecture
#
key modules

1. **the max pooling layer** 
    
    3 strategies:
    
    - sort input into a canonical order ⇒ 별로
    - treat the input as a sequence to train an RNN ⇒ 별로
    - **use a simple symmetric function to aggregate the information from each point** ⇒ 채택
    
2. **local and global information combination structure**
    - After computing the global point cloud feature vector, feed it back to per point features by concatenating the global feature with each of the point features → extract new per point feautres based on the combined point features ← is aware of both local, global information !!!
    
3. **two joint alignment networks**
    - align all input set to a canonical space before feature extraction
    - predict an affine transformation matrix by a mini-network and directly apply this transformation to the coordinates of input points
#
### Applications
#
<img width="403" alt="image" src="https://user-images.githubusercontent.com/98469609/204133698-fbd76d17-a790-4395-9add-85da3b6328c9.png">
1. 3D Object Classification
    - augment the point cloud on-the-fly by randomly rotation the obj along the up-axis and jitter the position of each points by a Gaussian noise with zero mean and 0.02 standard deviation
    - ModelNet40
2. 3D Object Part Segmentation
    - ShapeNet part data set
3. Semantic Segmentation in Scenes
    - Stanford 3D semantic parsing data set
#
### Architecture Design Analysis

- comparison with Alternative Order-invariant Methods
- Effectiveness of Input and Feature Transformations
- Robustness Test
#
### Visualizing PointNet

<img width="383" alt="image" src="https://user-images.githubusercontent.com/98469609/204133728-8e32e7ee-2f37-42dc-81c7-d5912af65bfd.png">

*critical point sets Cs*, (those contributed to the max pooled feature), summarizes the skeleton of the shape. The *upper-bound shapes Ns* illustrates the largest possible point cloud that give the same global shape feature f(S) as the input point cloud S.

→ losing some non-critical points does not change the global shape signature f(S) at all
#
### Time & Space Complexity Analysis
#
PointNet : space and time complexity : O(N)
#
able to process more than one million points per sec for point cloud classification or semantic segmentation with a 1080X GPU on TensorFlow → potential for real-time applications
#
### Conclusion

In this work, we propose a novel deep neural network PointNet that directly consumes point cloud. Our network provides a unified approach to a number of 3D recognition tasks including object classification, part segmentation and semantic segmentation, while obtaining on par or better results than state of the arts on standard benchmarks.
